packages/web3-utils/src/Utils.js

```diff
@@ -283,7 +283,7 @@ export const hexToNumber = (value) => {
 export const hexToNumberString = (value) => {
     if (!value) return value;
 
-    return toBN(value).toString(10);
+    return parseInt(value, 16).toString();
```

Comment:
This doesn't work because the given hex string could exceed the Number.MAX_VALUE of JS.

Reply:
The solution would be to remove the bignumber.js dependency and to use the native BigInt object with the combination of the JSBI polyfill.

Reply:
Yes right @nivida, I will remove the BN.js dependency and I will add BigInt with JSBI.

Reply:
should I remove BN.js dependecy from whole module or for only hexToNumber and hexToNumberString functions? @nivida

Reply:
BigInt and JSBI.BigInt is not working in all cases it gives the wrong result for "0000000000000000000000000000000000000000000000000000000000000010" hex string.

Reply:
@princesinha19 Hmm.. I will do some tests. I was just reading the specs until now and didn't test it. :)

Reply:
I found a repo which works fine in all cases. https://github.com/peterolson/BigInteger.js
Can I use that?

Reply:
I did test using this:
```
	hexToNumber = bigInt("75643564363473453456342378564387956906736546456235345", 16);	
	console.log("Using Big Int: ",hexToNumber)

	NormalParseInt = parseInt("75643564363473453456342378564387956906736546456235345", 16)
	console.log("Using Normal Parse Int: ",NormalParseInt)
```

**Output:**

![hexToNumber](https://user-images.githubusercontent.com/28489437/55642185-6d0dd700-57ed-11e9-87c8-ac0555e2b523.png)

Reply:
@nivida I think above repo is working fine and we can use that for our purpose. I verified it using https://www.rapidtables.com/convert/number/hex-to-decimal.html.

Reply:
@nivida did you got a chance to look to the above repo?

Reply:
did you got a chance to look into this @nivida?

Reply:
RE `JSBI.BigInt("0x0000000000000000000000000000000000000000000000000000000000000010")`

If you leave off the `0x` you will get `10` as the result, which is correct.

Reply:
No, I checked with 'Ox'. At was giving me 10 and that was wrong. I matched the result online. After that I didn't checked the JSBI. So, I need to check again that is it still an issue or it got fixed.

Reply:
@davidmurdoch The deimal of hex value `10` is `16` right? So, If I will leave `0x` then, also it should give 16. But I did check it with '0x' but it gave me the incorrect result. I remember at that time, I verified the result using https://www.rapidtables.com/convert/number/hex-to-decimal.html. But, I need to check it again that is it solved or still it's incorrect.

Reply:
If you leave off the `0x` you should get `10`, since the string `"000...00010"` always parses to `10` (when there is no base prefix specified the string will be treated as if it is a base 10 number).

`JSBI.BigInt` also supports binary encoding as `"0b10"` which should return `2` (`JSBI.BigInt("0b10")`), and octal as `"0o10"` which should be `8`. Octal is a confusing one because the _number_ `010` evaluates to `8`, but the _string_ `"010"` parsed as a `number` or a `JSBI.BigInt` is `10`.

Reply:
Yes right, @davidmurdoch. I checked it now and now, it's giving me the correct result for string with `0x` prefix.
But I don't know why it parses the string `"10"` as a number.

Reply:
Maybe now we could use JSBI. But the only problem is that it parses some string as base 10 number.

Reply:
It parses the string `"10"` as the number `10` because that is how both the JavaScript `number type and the new `bigint` type have been specced:

>  If radix is undefined or 0, it is assumed to be 10 except when the number begins with the code unit pairs 0x or 0X, in which case a radix of 16 is assumed.
https://tc39.github.io/ecma262/#sec-parseint-string-radix

Reply:
Ya, thanks for clarification @davidmurdoch. I got your point. Now, I think it will be a good option to use JSBI to solve all issues related to BN.